src/messages/MClientRequest.h

```diff
@@ -160,7 +162,17 @@ class MClientRequest : public Message {
 
   void decode_payload() {
     bufferlist::iterator p = payload.begin();
-    ::decode(head, p);
+
+    if (header.version >= 4) {
+      struct ceph_mds_request_head_new	new_mds_head;
+
+      ::decode(new_mds_head, p);
+      memcpy(&head, &(new_mds_head.oldest_client_tid), sizeof(head));
```

Comment:
Understood. I felt it was pretty icky too. I'll see about switching this around handle it that way.